src/hardware/pcspeaker.cpp

```diff
@@ -274,34 +277,28 @@ void PCSPEAKER_SetType(Bitu mode)
 	spkr.chan->Enable(true);
 }
 
-// Once the speaker is idle (as defined when num_actions is empty),
-// give it a short grace-time before ramping the volume down.
-// By keeping this idle-phase relatively short, we reduce the chance
-// PC-Speaker's DC-offset can influence other channels (such as the
-// Sound blaster or GUS).  Counts are in milliseconds.
-static void FadeVolume(uint16_t num_actions)
+// After the speaker has idled (as defined when the number of speaker
+// movements is zero) for a short time, wind down the DC-offset, and
+// then halt the channel.
+static void PlayOrSilence(const uint16_t speaker_movements,
```

Comment:
Suggestion: maybe `PlayOrFadeout` name would be more accurate description of what's this function is doing?

Reply:
I like it! replaced.